### PR TITLE
Adds Observer HUD buttons + Refactor of some ghost verbs

### PR DESCRIPTION
**Pull request by Arkatos1**
**What does this PR do:**
This PR adds Observer HUD buttons for Jump to Mob, Orbit, Re-enter corpse, and Teleport ghost verbs and refactors them (so they are even properly usable for the HUD buttons).

This will improve observing experience, as players do not have to clumsily navigate Ghost tab for the most commonly used ghost verbs, such as Orbit or Teleport.

**Images of sprite/map changes (IF APPLICABLE):**
ghostHud

Ported from /tg/station13.

Tested locally without any issue.

EDIT: I forgot to say that these icons are also animated. Animation plays when players hovers over them. Showcase:

<a href="https://gyazo.com/f6e37c47319cc4814b718b9574f842b5">Image from Gyazo</a>

**Changelog:**
:cl: Arkatos
add: Added Observer HUD buttons for Jump to Mob, Orbit, Re-enter corpse, and Teleport
imageadd: Added custom icons for Observer HUD buttons
/:cl:

### CODE REVIEW

File: code/modules/mob/dead/observer/observer.dm [hunk 1]
```diff
@@ -484,24 +486,29 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
...
-	if(istype(usr, /mob/dead/observer)) //Make sure they're an observer!
+	if(isobserver(usr)) //Make sure they're an observer!
+		var/list/dest = list() //List of possible destinations (mobs)
+		var/target = null	   //Chosen target.
 
-		if(!target)//Make sure we actually have a target
+		dest += getpois(mobs_only=1) //Fill list, prompt user with list
+		target = input("Please, select a mob!", "Jump to Mob", null, null) as null|anything in dest
+
+		if (!target)//Make sure we actually have a target
```
Comment (Ty-Omaha) [inline code/modules/mob/dead/observer/observer.dm]: no space after if

Comment (Arkatos1) [reply code/modules/mob/dead/observer/observer.dm]: fixed

File: code/modules/mob/dead/observer/observer.dm [hunk 2]
```diff
@@ -498,7 +498,7 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 		dest += getpois(mobs_only=1) //Fill list, prompt user with list
 		target = input("Please, select a mob!", "Jump to Mob", null, null) as null|anything in dest
 
-		if (!target)//Make sure we actually have a target
+		if (!target) //Make sure we actually have a target
```
Comment (Ty-Omaha) [inline code/modules/mob/dead/observer/observer.dm]: I meant: `if(!target)`

Comment (Arkatos1) [reply code/modules/mob/dead/observer/observer.dm]: fixed for real this time

**Comment by Kyep** [comment]
Can we use non-clown sprites for these?

The clown sprites kinda remind me of low-RP servers.

**Comment by Twinmold93** [comment]
What would you suggest instead as the sprite? Traditional greytide? Officer? Scientist? I believe the clown sprite was used partly because there is nothing that defines the actual character (other than it's a clown), but also because the clown is kind of a staple of SS13 as it is. I'd want to avoid, if it's switched, it just being standard greytide. Perhaps make it so they aren't all the same job/role? One as an officer, once as a chaplain, etc.

**Comment by Kyep** [comment]
Why can't they just have text labels, like "->MOB", "ORBIT", "BODY" and "TELE"?
Those are perfectly clear as to what they do, and easy to make.

If actual icons are needed...
JUMP = arrow pointing at a mob
ORBIT = arrow going around in a circle
BODY = corpse with the skull icon
TELE = dashed speed arrow

**Comment by taukausanake** [comment]
I vote generic assistant. You don't see them anymore so might be nice to have a spot for them

**Comment by Fox-McCloud** [comment]
I'm fine with the sprites. The clown is a core theme of SS13 culture, and has been on even paradise since we first launched.

**Comment by Necaladun** [comment]
For me it's more the bright colours of the clown, something more subtle like the ghost would be better.

Not really fussed though.

**Comment by Mitchs98** [comment]
Agreeing with Fox. Clown is definitely a staple in SS13. It doesn't remind me of LowRP servers, it just reminds me of SS13. Nothing more, nothing less.

Pretty sure a fair bit of us think SS13 everytime we hear the word clown or see one :P.

**Comment by Kyep** [comment]
To be clear, I'm objecting to this PR until the sprites are changed.
Once the sprites are changed, its fine. Good even. But the sprites must be changed.

**Comment by Tayyyyyyy** [comment]
Icons usually aren't that helpful anyways unless they're familiar ones (like home or font). UX wise, text buttons are probably better. People will get used to the icons eventually, but that's because they'll learn them, not because these icons are informative right off the bat. I'm still not sure what the black hole icon is supposed to do even after reading the PR. There are tooltips, obviously, but if your user interface requires tooltips to convey meaning, why not just use the text in the first place?

The clown icon does have the advantage of being gender neutral, though. I would prefer that over "generic assistant" because "generic" is kind of hard to define with such a diverse cast of characters.

**Comment by SkeletalElite** [comment]
> The clown icon does have the advantage of being gender neutral, though. I would prefer that over "generic assistant" because "generic" is kind of hard to define with such a diverse cast of characters.

Generic assistant= grey jumpsuit, insulated gloves, gas mask, cone or bucket on head. toolbox in hand black and white shoes.

**Comment by taukausanake** [comment]
> Generic assistant *snip

That's a little too much though, and not all assistants have all that. Just a plain baldy might be okay as that is a generic as you can get. The default no-change settings for a new player, that's almost as iconic an image, if not hated, as the clown

**Comment by Arkatos1** [comment]
I forgot to mention these icons are also animated, main post updated with a showcase of them.

**Comment by taukausanake** [comment]
> animated icons

Speedmerge now

File: code/_onclick/hud/_defines.dm [hunk 3]
```diff
@@ -154,6 +154,12 @@
 #define ui_bot_radio "EAST-1:28,SOUTH:7"
 #define ui_bot_pull "EAST-2:26,SOUTH:7"
 
+//Ghosts
+#define ui_ghost_jumptomob "SOUTH:6,CENTER-2:24"
+#define ui_ghost_orbit "SOUTH:6,CENTER-1:24"
+#define ui_ghost_reenter_corpse "SOUTH:6,CENTER:24"
+#define ui_ghost_teleport "SOUTH:6,CENTER+1:24"
+
```
Comment (Fox-McCloud) [inline code/_onclick/hud/_defines.dm]: eventually we should do the pAI thing too.

Comment (Arkatos1) [reply code/_onclick/hud/_defines.dm]: Absolutely agree, but that would mean tinkering with how pAIs are chosen, so I decided to move to the future PR(s).

Comment (Fox-McCloud) [reply code/_onclick/hud/_defines.dm]: Yeah, that's fine. This is more important and allows @AffectedArc07 to move forward with Parallax.

**Pull request merged**